src/meltano/core/tracking/plugins.py

```diff
@@ -0,0 +1,57 @@
+"""Tracking plugin context for the Snowplow tracker."""
+from __future__ import annotations
+
+import uuid
+
+from snowplow_tracker import SelfDescribingJson
+from structlog.stdlib import get_logger
+
+from meltano.core.plugin.project_plugin import ProjectPlugin
+from meltano.core.utils import hash_sha256
+
+logger = get_logger(__name__)
+
+PLUGINS_CONTEXT_SCHEMA = "iglu:com.meltano/plugins_context/jsonschema"
+PLUGINS_CONTEXT_SCHEMA_VERSION = "1-0-0"
+
+
+def _from_plugin(plugin: ProjectPlugin, cmd: str) -> dict:
+    return {
+        "category": str(plugin.type),
+        "name_hash": hash_sha256(plugin["info"].get("name")),
+        # TODO: what if its "original" should be we hashing namespace and sending that as well?
+        "variant_name_hash": hash_sha256(plugin["info"].get("variant")),
```

Comment:
@aaronsteers in a lot of cases the variant is set to "original" which I assume is ok. Curious though if we should snag a hashed version of plugin["info"]["namespace"] as well ?